packages/medusa/src/api/routes/store/payment-collections/index.ts

```diff
@@ -33,20 +34,31 @@ export default (app, container) => {
   )
 
   route.post(
-    "/:id/authorize",
-    middlewares.wrap(require("./authorize-payment-collection").default)
+    "/:id/sessions",
+    transformBody(StorePaymentCollectionSessionsReq),
+    middlewares.wrap(require("./manage-payment-session").default)
   )
 
   route.post(
-    "/:id/sessions",
-    transformBody(StoreManagePaymentCollectionSessionRequest),
-    middlewares.wrap(require("./manage-payment-sessions").default)
+    "/:id/sessions/:session_id",
+    middlewares.wrap(require("./refresh-payment-session").default)
   )
 
   route.post(
-    "/:id/sessions/:session_id/refresh",
-    transformBody(StoreRefreshPaymentCollectionSessionRequest),
-    middlewares.wrap(require("./refresh-payment-session").default)
+    "/:id/sessions/:session_id/authorize",
+    middlewares.wrap(require("./authorize-payment-session").default)
+  )
+
+  route.post(
+    "/:id/batch/sessions",
+    transformBody(StorePostPaymentCollectionsBatchSessionsReq),
+    middlewares.wrap(require("./manage-batch-payment-sessions").default)
+  )
+
+  route.post(
+    "/:id/batch/sessions/authorize",
+    transformBody(StorePostPaymentCollectionsBatchSessionsAuthorizeReq),
+    middlewares.wrap(require("./authorize-batch-payment-sessions").default)
```

Comment:
If you change the order of the batch operations they will work. See this example: https://codesandbox.io/s/express-js-forked-ci0l54?file=/src/index.js

I.e. make sure that:

`.post("/:id/sessions/batch/authorize" ...` is defined before `.post("/:id/sessions/:session_id/authorize" ...`

Reply:
done. 😄 nobody will be able to use `batch` as a payment session id.